.github/workflows/nightly-goth.yml

```
@@ -12,11 +12,35 @@ on:
     - cron: '0 3 * * *'
 
 jobs:
+  prepare-matrix:
+    name: Prepare matrix JSON
+    runs-on: ubuntu-latest
+    outputs:
+      matrix-json: ${{ steps.get-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      # finds branches with names matching 'release\/v[0-9](\.[0-9])+' (e.g. release/v0.7) and returns one with highest version
+      - name: Get latest stable branch
+        id: latest-stable
+        # sed removes leading whitespaces and '*' characters (git uses it to indicate the current branch)
+        run: |
+          branch=$(git branch -a | sed 's/remotes\/origin\///' | sed 's/^[ \t*]*//' | grep -E '^release\/v[0-9](\.[0-9])+$' | sort -Vr | head -1)
```

In case `yagna` gets `v0.7.10`, `v1.23.4` or `v20.0.0` release :)
```suggestion
          branch=$(git branch -a | sed 's/remotes\/origin\///' | sed 's/^[ \t*]*//' | grep -E '^release\/v[0-9]+(\.[0-9]+)+$' | sort -Vr | head -1)
```

---

BTW, you can also replace `sed 's/remotes\/origin\///'` with slightly simpler `sed 's:remotes/origin::'`.